src/core/server/http/router/request.ts

```diff
@@ -186,11 +196,16 @@ export class KibanaRequest<
 
   private getEvents(request: Request): KibanaRequestEvents {
     const finish$ = merge(
-      fromEvent(request.raw.req, 'end'), // all data consumed
```

Comment:
The initial implementation was based on https://github.com/nodejs/node/issues/21705 discussion and asserted by test https://github.com/elastic/kibana/pull/55061#discussion_r368014273, showing that `Request.end` happens after `Response.finish`, even though it shouldn't. I might miss the fact that both were tested with `GET` requests.